fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/mixin/resource/loader/ResourcePackManagerMixin.java

```diff
@@ -65,4 +82,50 @@ public void construct(ResourcePackProvider[] resourcePackProviders, CallbackInfo
 			providers.add(new ModResourcePackCreator(ResourceType.SERVER_DATA));
 		}
 	}
+
+	@Inject(method = "buildEnabledProfiles", at = @At(value = "INVOKE", target = "Lcom/google/common/collect/ImmutableList;copyOf(Ljava/util/Collection;)Lcom/google/common/collect/ImmutableList;", shift = At.Shift.BEFORE))
+	private void handleAutoEnableDisable(Collection<String> enabledNames, CallbackInfoReturnable<List<ResourcePackProfile>> cir, @Local List<ResourcePackProfile> enabledAfterFirstRun) {
+		this.refreshAutoEnabledPacks(enabledAfterFirstRun);
+	}
+
+	@Inject(method = "enable", at = @At(value = "INVOKE", target = "Ljava/util/List;add(Ljava/lang/Object;)Z", shift = At.Shift.AFTER))
+	private void handleAutoEnable(String profile, CallbackInfoReturnable<Boolean> cir, @Local List<ResourcePackProfile> newlyEnabled) {
+		if (ModResourcePackCreator.POST_CHANGE_HANDLE_REQUIRED.contains(profile)) {
+			this.refreshAutoEnabledPacks(newlyEnabled);
+		}
+	}
+
+	@Unique
+	private void refreshAutoEnabledPacks(List<ResourcePackProfile> enabledProfiles) {
+		LOGGER.info("Started with: {}", enabledProfiles.stream().map(ResourcePackProfile::getName).toList());
+		enabledProfiles.removeIf(FabricResourcePackProfile::isHidden);
+		LOGGER.info("Removed internal packs: {}", enabledProfiles.stream().map(ResourcePackProfile::getName).toList());
+		ListIterator<ResourcePackProfile> it = enabledProfiles.listIterator();
+		// LinkedHashSet for debug log sort, will switch to normal HashSet once finished
+		Set<String> seen = new LinkedHashSet<>();
+
+		while (it.hasNext()) {
+			ResourcePackProfile profile = it.next();
+			seen.add(profile.getName());
+
+			for (ResourcePackProfile p : this.profiles.values()) {
+				if (p.isHidden() && p.parentsEnabled(seen) && seen.add(p.getName())) {
+					it.add(p);
+					LOGGER.info("cur @ {}, auto-enabled {}, currently enabled: {}", profile.getName(), p.getName(), seen);
```

Comment:
Nit: drop to debug level, see others as well.

Reply:
Yeah, they will be removed when it's done (or dropped to debug).